File: nodejs/com/inmobi/monetization/api/response/ad/NativeResponse.js
var AdResponse = require('./AdResponse.js');
var Utils = require('../../utils/Utils.js');

////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
// This class represents a Native Ad Response object
// @author Rishabh Chowdhary
////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
function NativeResponse() {
	AdResponse.apply(this,Array.prototype.slice.call(arguments));
	this.pubContent = null;
	this.namespace = null;
	this.contextCode = null;
}
NativeResponse.prototype.isValid = function() {
	if(Utils.isStringValid(this.pubContent) 
		&& Utils.isStringValid(this.contextCode)
		&& Utils.isStringValid(this.namespace)) {
		return true;
	}
	return false;
}
NativeResponse.prototype.convertPubContentToJSON = function() {
	if(Utils.isStringValid(this.pubContent)) {
		return new Buffer(this.pubContent, 'base64').toString('ascii');
	}
	return null;
}

module.exports = NativeResponse;

File: ruby/com/inmobi/monetization/api/response/NativeResponse.rb
require_relative('AdResponse')
require 'base64'

########################################################################
# This class represents a Native Ad Response object
# @author Rishabh Chowdhary
########################################################################
class NativeResponse < AdResponse

	def initialize
		@pubContent = nil
		@namespace = nil
		@contextCode = nil
	end
	public
	attr_accessor :pubContent
	attr_accessor :namespace
	attr_accessor :contextCode

	# Returns a base64 decode JSON String of the pubContent
	#NOTE: This should be the same exact JSON template as
	# setup by you on the InMobi dashboard for a Native Ad.
	def convertPubContentToJSON
		if(Utils.isStringValid(@pubContent))
			return Base64.decode64(@pubContent);
		end
		return nil
	end

	def isValid
		if(Utils.isStringValid(@pubContent) && Utils.isStringValid(@namespace) && Utils.isStringValid(@contextCode) )
			return true
		end
		return false
	end
end